sleap_io/model/skeleton.py

```diff
@@ -6,8 +6,15 @@
 """
 
 from __future__ import annotations
-from attrs import define
-from typing import Optional
+from attrs import define, field
+from typing import Any, Dict, Iterable, List, Optional, Tuple, Union, Text
+from itertools import count
+
+# import networkx as nx
+import numpy as np
+
+NodeRef = Union[str, "Node"]
+skeleton_idx = count(0)
 
 
 @define
```

Comment:
Lets specify `@define(auto_attrib=True)` as this will force us to type hint each attribute per:
> auto_attribs ([bool](https://docs.python.org/3/library/functions.html#bool)) –
 If True, collect [PEP 526](https://www.python.org/dev/peps/pep-0526/)-annotated attributes (Python 3.6 and later only) from the class body.
 In this case, you must annotate every field. If attrs encounters a field that is set to an [attr.ib](https://www.attrs.org/en/stable/api.html#attr.ib) but lacks a type annotation, an [attr.exceptions.UnannotatedAttributeError](https://www.attrs.org/en/stable/api.html#attrs.exceptions.UnannotatedAttributeError) is raised.